packages/client/src/components/MovieCard/MovieCard.jsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import { Card, CardMedia, Typography, Rating, Box } from '@mui/material';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import useFavorites from './useFavorites'; // created a custom hook to manage favorites
+
+const BasicRating = () => {
+  return (
+    <Rating
+      name="read-only"
+      value={4}
+      readOnly
+      sx={{
+        position: 'relative',
+        left: '1.96rem',
+        borderRadius: '0.59',
+        maxHeight: 'min-content',
+        width: '3rem',
+        top: '2rem',
+        paddingBottom: '4rem',
+        color: '#003E2F',
+      }}
+    />
+  );
+};
+
+export default function MovieCard(props) {
+  const [favorites, toggleFavorite] = useFavorites([]);
+  const isFavorite = favorites.includes(props.movie);
+
+  return (
+    <Card
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        maxHeight: 'min-content',
+        width: '17.43rem',
+        left: '6rem',
+        background:
+          'linear-gradient(166.18deg, rgba(0, 0, 0, 0.4) 0%, rgba(0, 0, 0, 0) 98.81%), #00FFC2',
+        boxShadow: '20px 40px 70px rgba(0, 0, 0, 0.25)',
+      }}
+    >
+      <Box
+        sx={{
+          position: 'relative',
+        }}
+      >
+        {isFavorite ? (
+          <FavoriteIcon
+            sx={{
+              display: 'block',
+              position: 'absolute',
+              width: '1.9rem',
+              height: '1.75rem',
+              top: '2.25rem',
+              right: '1.96rem',
+              color: '#FF0000',
+              padding: '1rem',
+              backgroundColor: '#000000CC',
+              zIndex: 1,
+              '&:hover': {
+                color: '#FF0000',
+                cursor: 'pointer',
+              },
+            }}
+            onClick={() => toggleFavorite(props.movie)}
+          />
+        ) : (
+          <FavoriteBorderIcon
+            sx={{
+              display: 'block',
+              position: 'absolute',
+              width: '1.9rem',
+              height: '1.75rem',
+              top: '2.25rem',
+              right: '1.96rem',
+              color: '#FFFFFF',
+              padding: '1rem',
+              backgroundColor: '#000000CC',
+              zIndex: 1,
+              '&:hover': {
+                color: '#FF0000',
+                cursor: 'pointer',
+              },
+            }}
+            onClick={() => toggleFavorite(props.movie)}
+          />
+        )}
```

Comment:
or you could make just the backgorundColor conditional style dependent on `isFavourite` and then you would not need to copy-paste logic :)

Reply:
yeah, i tried to do it but unfortunately background color is taking as square, i couldn't find the way only do it inside the heart and when i checked MUI there are 2 different heart icons so I did this way but now i shortened the code and put styles in a different variable. 
<img width="433" alt="Screenshot 2023-05-13 at 13 40 11" src="https://github.com/HackYourFuture-CPH/CINEMANIA/assets/113375790/92f01d5e-6451-4510-9270-e63ef037e058">

Reply:
oooh I didn't notice those were two different icons! Anyway putting the common styles into a var can be a good idea :)